cli/doc/parser.rs

```diff
@@ -22,43 +23,96 @@ use crate::swc_ecma_parser::Session;
 use crate::swc_ecma_parser::SourceFileInput;
 use crate::swc_ecma_parser::Syntax;
 use crate::swc_ecma_parser::TsConfig;
+
+use deno_core::ErrBox;
+use deno_core::ModuleSpecifier;
+use futures::Future;
 use regex::Regex;
+use std::collections::HashMap;
+use std::error::Error;
+use std::fmt;
+use std::pin::Pin;
 use std::sync::Arc;
 use std::sync::RwLock;
 
+use super::namespace::NamespaceDef;
+use super::node;
+use super::node::ModuleDoc;
 use super::DocNode;
 use super::DocNodeKind;
 use super::Location;
 
-pub type SwcDiagnostics = Vec<Diagnostic>;
+#[derive(Clone, Debug)]
+pub struct SwcDiagnosticBuffer {
+  pub diagnostics: Vec<Diagnostic>,
+}
```

Comment:
I don't really see how SwcDiagnosticBuffer is related to this patch.. but whatever.

Reply:
Me neither, this is just some infrastructure needed for `swc`, just renamed to something more descriptive